### PR TITLE
Add capi monitor to update health_status

### PR DESCRIPTION
**Pull request by dalees**
A WIP status monitor that uses CAPI state only.

### CODE REVIEW

**Comment by dalees** [comment]
@mkjpryor This could use some discussion as this new monitor class currently uses internal functions from the Driver as this is the only place they exist. I **don't** want to pass driver into this class ;)

Do you have thoughts on refactoring the driver so these can be commonly accessed by multiple classes? Happy to pull these out an implement the changes, but wanted to make sure I showed the reason and sample code before I went ahead with it!

Driver functions:
* Static function `_namespace()`
* Class variable `_k8s_client`
* I don't use `_get_chart_release_name()` or `._sanitized_name()` (but should)

**Comment by mkjpryor** [comment]
@dalees I definitely think it makes sense to move those into a utils module or similar rather than re-implementing them

File: magnum_capi_helm/common/capi_monitor.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+            oslo.versionedobjects/blob/master/oslo_versionedobjects/fields.py
+
+        :return: None.
+
+        The class variable data is updated with current status and reason.
+
+        """
+        resource_name = self.cluster.stack_id
+        # TODO(dalees): Don't access driver internals!
+        namespace = self.driver._namespace(self.cluster)
```
Comment (JohnGarbutt) [inline magnum_capi_helm/common/capi_monitor.py:75]: Yeah, lets move those as module methods in `magnum_capi_helm/kubernetes.py`

**Review by JohnGarbutt — CHANGES_REQUESTED** [review]

File: magnum_capi_helm/common/capi_monitor.py (hunk 1 above)
Comment (dalees) [reply magnum_capi_helm/common/capi_monitor.py]: Moved these shared static functions into `magnum_capi_helm/driver_utils.py`.

This cleaned up the monitor from having to use a driver reference, so this is removed.

**Comment by mkjpryor** [comment]
This looks great to me. Personally I would like to see some unit tests for the monitor - @JohnGarbutt what do you think?

**Comment by dalees** [comment]
> This looks great to me. Personally I would like to see some unit tests for the monitor - @JohnGarbutt what do you think?

Yeah, good point. I've added a few unit tests in `tests/common/test_capi_monitor.py`

**Pull request merged**